ucp/_libs/ucxio.pyx

```diff
@@ -41,12 +41,20 @@ class UCXIO(RawIOBase):
         req = self.rkey.ep.flush(blocking_handler, cb_args=(self.cb_finished,))
         self.block_on_request(req)
 
-    def seek(self, pos, whence=0):
-        if whence == 1:
+    def seek(self, pos, whence=SEEK_SET):
+        if whence == SEEK_SET:
+            self.pos = min(max(pos, 0), self.length)
+        elif whence == SEEK_CUR:
+            if pos < 0:
+                self.pos = max(self.pos + pos, 0)
+            else:
+                self.pos = min(self.pos + pos, self.length)
             pos += self.pos
-        if whence == 2:
-            pos = self.length - pos
-        self.pos = pos
+        elif whence == SEEK_END:
+            pos = min(max(self.poss + pos, 0), self.length)
```

Comment:
Def should add tests for this, like with the above `poss` problem. May take a bit since it's a short week here because of the US holiday and the UCF conference next week.

Reply:
No rush, please take your time. 🙂